tests/classes/json_encode_non_serializable_deprecated.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+json_encode() on instances of classes market with ZEND_ACC_NOT_SERIALIZABLE
```

Comment:
This test should be located in the corresponding tests directory for json extension, `ext/json/tests`. And despite the outcome of the discussion on the internal list, I would see the benefit of including the tests now. I was surprised that those paths were not tested already.